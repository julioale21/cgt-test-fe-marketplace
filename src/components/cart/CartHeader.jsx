import React from 'react';
import PropTypes from 'prop-types';
import { Box, Typography, CardHeader } from '@mui/material';
import { OutlinedButton } from '../buttons/OutlinedButton';


const CartHeader = ({ itemCount, onClearCart }) => (
  <CardHeader
    title={
      <Box sx={{ display: 'flex', justifyContent: 'space-between', alignItems: 'center' }}>
        <Typography variant="h6">Shopping Cart ({itemCount} items)</Typography>
        <OutlinedButton size="small" onClick={onClearCart}>
          Clear Cart
        </OutlinedButton>
      </Box>
    }
  />
);

CartHeader.propTypes = {
  itemCount: PropTypes.number.isRequired,
  onClearCart: PropTypes.func.isRequired
};

export { CartHeader };
