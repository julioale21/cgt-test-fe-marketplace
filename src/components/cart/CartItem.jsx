import React from 'react';
import PropTypes from 'prop-types';
import { Box, Typography, IconButton } from '@mui/material';
import { Add as AddIcon, Remove as RemoveIcon, Delete as DeleteIcon } from '@mui/icons-material';

const CartItem = ({ item, onIncrement, onDecrement, onRemove }) => (
  <Box
    sx={{
      py: 3,
      display: 'flex',
      gap: 3
    }}
  >
    <Box
      component="img"
      src={item.image}
      alt={item.name}
      sx={{
        width: '80px',
        height: '80px',
        objectFit: 'cover',
        borderRadius: 2
      }}
    />

    <Box
      sx={{
        flexGrow: 1,
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'space-between'
      }}
    >
      <Box
        sx={{
          display: 'flex',
          justifyContent: 'space-between',
          alignItems: 'flex-start'
        }}
      >
        <Box>
          <Typography variant="h6" sx={{ fontWeight: 500 }}>
            {item.name}
          </Typography>
          <Typography variant="h5" sx={{ mt: 1, fontWeight: 500 }}>
            ${item.price.toFixed(2)}
          </Typography>
        </Box>

        <Box
          sx={{
            display: 'flex',
            alignItems: 'center',
            gap: 1
          }}
        >
          <IconButton
            onClick={onDecrement}
            size="small"
            sx={{
              color: 'action.active',
              '&:hover': {
                color: 'primary.main',
                bgcolor: 'primary.lighter'
              }
            }}
          >
            <RemoveIcon />
          </IconButton>
          <Typography
            sx={{
              minWidth: '32px',
              textAlign: 'center'
            }}
          >
            {item.quantity}
          </Typography>
          <IconButton
            onClick={onIncrement}
            size="small"
            sx={{
              color: 'action.active',
              '&:hover': {
                color: 'primary.main',
                bgcolor: 'primary.lighter'
              }
            }}
          >
            <AddIcon />
          </IconButton>
          <IconButton
            onClick={onRemove}
            size="small"
            color="error"
            sx={{
              ml: 2,
              '&:hover': {
                bgcolor: 'error.lighter'
              }
            }}
          >
            <DeleteIcon />
          </IconButton>
        </Box>
      </Box>
    </Box>
  </Box>
);

CartItem.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    image: PropTypes.string.isRequired,
    quantity: PropTypes.number.isRequired
  }).isRequired,
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
  onRemove: PropTypes.func.isRequired
};

export { CartItem };
