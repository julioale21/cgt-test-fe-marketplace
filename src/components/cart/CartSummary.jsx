import React from 'react';
import { Box, Card, CardContent, CardHeader, Typography, Divider, Stack } from '@mui/material';
import PropTypes from 'prop-types';

import {
  ArrowBack as ArrowBackIcon,
  CreditCard as CreditCardIcon,
  LocalShipping as ShippingIcon
} from '@mui/icons-material';

import {
  CartSummaryRow,
  ContainedButton,
  OutlinedButton,
  PaymentMethodIcon
} from '../../components';

const CartSummary = ({
  subtotal,
  shipping = 0,
  taxes,
  finalTotal,
  onCheckout,
  onContinueShopping
}) => {
  const formatAmount = (amount) => {
    const nonNegativeAmount = Math.max(0, amount);
    return `$${nonNegativeAmount.toLocaleString('en-US', {
      minimumFractionDigits: 2,
      maximumFractionDigits: 2
    })}`;
  };

  return (
    <Card sx={{ position: { md: 'sticky' }, top: { md: '24px' } }}>
      <CardHeader title="Order Summary" />
      <Divider />
      <CardContent>
        <Stack spacing={2}>
          <CartSummaryRow label="Subtotal" value={formatAmount(subtotal)} />
          <CartSummaryRow
            label="Shipping"
            value={shipping === 0 ? 'Free' : formatAmount(shipping)}
          />
          <CartSummaryRow label="Estimated tax" value={formatAmount(taxes)} />
          <Divider />
          <CartSummaryRow
            label="Total"
            value={formatAmount(finalTotal)}
            variant="h6"
            labelColor="text.primary"
          />

          <ContainedButton
            fullWidth
            onClick={onCheckout}
            startIcon={<CreditCardIcon />}
            sx={{ py: 1.5 }}
          >
            Continue to Checkout
          </ContainedButton>

          <OutlinedButton
            fullWidth
            onClick={onContinueShopping}
            startIcon={<ArrowBackIcon />}
            sx={{ py: 1.5 }}
          >
            Continue Shopping
          </OutlinedButton>

          <Box sx={{ mt: 2 }}>
            <Typography variant="subtitle2" gutterBottom sx={{ mb: 1 }}>
              Secure Checkout
            </Typography>
            <Stack
              direction="row"
              spacing={2}
              sx={{
                py: 2,
                px: 1,
                bgcolor: 'grey.50',
                borderRadius: 1,
                justifyContent: 'center'
              }}
            >
              <PaymentMethodIcon
                src="https://cdn.visa.com/v2/assets/images/logos/visa/blue/logo.png"
                alt="Visa"
              />
              <PaymentMethodIcon
                src="https://cdn.worldvectorlogo.com/logos/mastercard-7.svg"
                alt="Mastercard"
              />
              <PaymentMethodIcon
                src="https://www.paypalobjects.com/digitalassets/c/website/logo/full-text/pp_fc_hl.svg"
                alt="PayPal"
              />
            </Stack>
          </Box>

          <Box sx={{ mt: 2 }}>
            <Typography
              variant="body2"
              color="text.secondary"
              sx={{ display: 'flex', alignItems: 'center', gap: 1 }}
            >
              <ShippingIcon fontSize="small" />
              Free shipping on all orders
            </Typography>
          </Box>
        </Stack>
      </CardContent>
    </Card>
  );
};

CartSummary.propTypes = {
  subtotal: PropTypes.number.isRequired,
  shipping: PropTypes.number,
  taxes: PropTypes.number.isRequired,
  finalTotal: PropTypes.number.isRequired,
  onCheckout: PropTypes.func.isRequired,
  onContinueShopping: PropTypes.func.isRequired
};

export { CartSummary };
