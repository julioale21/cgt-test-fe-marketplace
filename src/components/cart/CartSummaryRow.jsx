import React from 'react';
import PropTypes from 'prop-types';
import { Box, Typography } from '@mui/material';

const CartSummaryRow = ({
  label,
  value,
  variant = 'body1',
  labelColor = 'text.secondary',
  valueColor = 'text.primary'
}) => {
  return (
    <Box sx={{ display: 'flex', justifyContent: 'space-between' }}>
      <Typography color={labelColor} variant={variant}>
        {label}
      </Typography>
      <Typography color={valueColor} variant={variant}>
        {value}
      </Typography>
    </Box>
  );
};

CartSummaryRow.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  variant: PropTypes.oneOf([
    'body1',
    'body2',
    'h1',
    'h2',
    'h3',
    'h4',
    'h5',
    'h6',
    'subtitle1',
    'subtitle2',
    'caption',
    'overline'
  ]),
  labelColor: PropTypes.string,
  valueColor: PropTypes.string
};

export { CartSummaryRow };
