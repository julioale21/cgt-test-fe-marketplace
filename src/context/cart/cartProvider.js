import { useReducer, useCallback } from 'react';
import { cartReducer } from './cartReducer';
import { initialState, CART_ACTIONS } from './types';
import { CartContext } from './cartContext';

export const CartProvider = ({ children }) => {
  const [state, dispatch] = useReducer(cartReducer, initialState);

  const addToCart = useCallback(({ product, quantity }) => {
    if (!product || product.price === undefined || quantity === undefined) {
      throw new Error("Invalid product or quantity passed to addToCart");
    }
    dispatch({ 
      type: CART_ACTIONS.ADD_TO_CART, 
      payload: { product, quantity }
    });
  }, []);

  const removeFromCart = useCallback((productId) => {
    dispatch({ 
      type: CART_ACTIONS.REMOVE_FROM_CART, 
      payload: productId 
    });
  }, []);

  const updateQuantity = useCallback((productId, quantity) => {
    dispatch({
      type: CART_ACTIONS.UPDATE_QUANTITY,
      payload: { id: productId, quantity }
    });
  }, []);

  const clearCart = useCallback(() => {
    dispatch({ type: CART_ACTIONS.CLEAR_CART });
  }, []);

  const value = {
    items: state.items,
    total: state.total,
    addToCart,
    removeFromCart,
    updateQuantity,
    clearCart
  };

  return (
    <CartContext.Provider value={value}>
      {children}
    </CartContext.Provider>
  );
};
