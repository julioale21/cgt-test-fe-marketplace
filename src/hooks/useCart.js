import { useContext } from 'react';
import { CartContext } from '../context/cart/cartContext';


export const useCart = () => {
  const context = useContext(CartContext);
  
  if (context === undefined) {
    throw new Error('useCart debe usarse dentro de un CartProvider');
  }
  
  return context;
};