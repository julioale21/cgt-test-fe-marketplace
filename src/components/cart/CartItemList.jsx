import React from 'react';
import PropTypes from 'prop-types';
import { Stack } from '@mui/material';
import { CartItem } from './CartItem';

const CartItemList = ({ items, onIncrement, onDecrement, onRemove }) => (
  <Stack spacing={2}>
    {items.map((item) => (
      <CartItem
        key={item.id}
        item={item}
        onIncrement={() => onIncrement(item)}
        onDecrement={() => onDecrement(item)}
        onRemove={() => onRemove(item.id)}
      />
    ))}
  </Stack>
);

CartItemList.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      price: PropTypes.number.isRequired,
      image: PropTypes.string.isRequired,
      quantity: PropTypes.number.isRequired
    })
  ).isRequired,
  onIncrement: PropTypes.func.isRequired,
  onDecrement: PropTypes.func.isRequired,
  onRemove: PropTypes.func.isRequired
};

export { CartItemList };
