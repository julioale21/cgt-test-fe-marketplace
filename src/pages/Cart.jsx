import React from 'react';
import { CartItemList, CartSummary, EmptyCart } from '../components';
import { Box, Typography, Button, Grid } from '@mui/material';
import { RemoveShoppingCart as ClearCartIcon } from '@mui/icons-material';
import { useCheckout } from '../hooks/useCheckout';

const Cart = () => {
  const {
    items,
    cartItemsCount,
    isEmpty,
    subtotal,
    shipping,
    taxes,
    finalTotal,
    handleIncrement,
    handleDecrement,
    handleCheckout,
    handleContinueShopping,
    clearCart,
    removeFromCart
  } = useCheckout();

  if (isEmpty) {
    return <EmptyCart onContinueShopping={handleContinueShopping} />;
  }

  return (
    <Box
      sx={{
        maxWidth: 1400,
        mx: 'auto',
        px: { xs: 2, md: 4 },
        py: { xs: 3, md: 5 }
      }}
    >
      <Grid container spacing={4}>
        <Grid item xs={12} md={8}>
          <Box
            sx={{
              mb: 3,
              display: 'flex',
              justifyContent: 'space-between',
              alignItems: 'center'
            }}
          >
            <Typography variant="h4" sx={{ fontWeight: 600 }}>
              {`Shopping Cart (${cartItemsCount} items)`}
            </Typography>
            <Button
              onClick={clearCart}
              startIcon={<ClearCartIcon />}
              color="secondary"
              sx={{
                '&:hover': {
                  bgcolor: 'secondary.lighter'
                }
              }}
            >
              Clear Cart
            </Button>
          </Box>

          <Box sx={{ mt: 3 }}>
            <CartItemList
              items={items}
              onIncrement={handleIncrement}
              onDecrement={handleDecrement}
              onRemove={removeFromCart}
            />
          </Box>
        </Grid>

        <Grid item xs={12} md={4}>
          <CartSummary
            subtotal={subtotal}
            shipping={shipping}
            taxes={taxes}
            finalTotal={finalTotal}
            onCheckout={handleCheckout}
            onContinueShopping={handleContinueShopping}
          />
        </Grid>
      </Grid>
    </Box>
  );
};

export default Cart;
