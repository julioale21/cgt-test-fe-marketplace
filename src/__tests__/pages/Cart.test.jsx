import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import * as useCartHook from '../../hooks/useCart';
import { CartProvider } from '../../context/cart/cartProvider';
import Cart from '../../pages/Cart';

jest.mock('../../hooks/useCart');
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn()
}));

describe('Cart', () => {
  const mockNavigate = jest.fn();
  const mockUpdateQuantity = jest.fn();
  const mockRemoveFromCart = jest.fn();
  const mockClearCart = jest.fn();

  const mockItems = [
    {
      id: 1,
      name: 'Test Product',
      price: 29.99,
      quantity: 2,
      image: 'test.jpg'
    }
  ];

  const renderCart = (mockItems = []) => {
    jest.spyOn(require('react-router-dom'), 'useNavigate').mockImplementation(() => mockNavigate);

    useCartHook.useCart.mockReturnValue({
      items: mockItems,
      total: mockItems.reduce((acc, item) => acc + item.price * item.quantity, 0),
      removeFromCart: mockRemoveFromCart,
      updateQuantity: mockUpdateQuantity,
      clearCart: mockClearCart,
      cartItemsCount: mockItems.reduce((acc, item) => acc + item.quantity, 0),
      isEmpty: mockItems.length === 0
    });

    return render(
      <MemoryRouter>
        <CartProvider>
          <Cart />
        </CartProvider>
      </MemoryRouter>
    );
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot with empty cart', () => {
    const { container } = renderCart([]);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot with items in cart', () => {
    const { container } = renderCart(mockItems);
    expect(container).toMatchSnapshot();
  });

  it('should render EmptyCart component when cart is empty', () => {
    renderCart([]);
    expect(screen.getByText('Your cart is empty')).toBeInTheDocument();
  });

  it('should render cart content when items exist', () => {
    renderCart(mockItems);
    const cartItemsCount = mockItems.reduce((acc, item) => acc + item.quantity, 0);
    expect(screen.getByText(`Shopping Cart (${cartItemsCount} items)`)).toBeInTheDocument();
    expect(screen.getByText(mockItems[0].name)).toBeInTheDocument();
  });

  it('should call updateQuantity when increment button is clicked', () => {
    renderCart(mockItems);
    const incrementButton = screen.getByTestId('AddIcon').parentElement;
    fireEvent.click(incrementButton);
    expect(mockUpdateQuantity).toHaveBeenCalledWith(mockItems[0].id, mockItems[0].quantity + 1);
  });

  it('should call updateQuantity when decrement button is clicked and quantity > 1', () => {
    renderCart(mockItems);
    const decrementButton = screen.getByTestId('RemoveIcon').parentElement;
    fireEvent.click(decrementButton);
    expect(mockUpdateQuantity).toHaveBeenCalledWith(mockItems[0].id, mockItems[0].quantity - 1);
  });

  it('should call removeFromCart when decrement button is clicked and quantity is 1', () => {
    const itemWithQuantityOne = [
      {
        ...mockItems[0],
        quantity: 1
      }
    ];
    renderCart(itemWithQuantityOne);
    const decrementButton = screen.getByTestId('RemoveIcon').parentElement;
    fireEvent.click(decrementButton);
    expect(mockRemoveFromCart).toHaveBeenCalledWith(itemWithQuantityOne[0].id);
  });

  it('should call clearCart when clear cart button is clicked', () => {
    renderCart(mockItems);
    fireEvent.click(screen.getByText('Clear Cart'));
    expect(mockClearCart).toHaveBeenCalled();
  });

  it('should navigate to checkout when checkout button is clicked', () => {
    renderCart(mockItems);

    fireEvent.click(screen.getByText('Continue to Checkout'));

    expect(mockNavigate).toHaveBeenCalledWith('/checkout');
  });

  it('should navigate to home when continue shopping is clicked', () => {
    renderCart(mockItems);
    fireEvent.click(screen.getByText('Continue Shopping'));
    expect(mockNavigate).toHaveBeenCalledWith('/');
  });

  it('should calculate correct totals', () => {
    renderCart(mockItems);
    const subtotal = mockItems[0].price * mockItems[0].quantity;
    const taxes = subtotal * 0.1;
    const finalTotal = subtotal + taxes;

    expect(screen.getByText(`$${subtotal.toFixed(2)}`)).toBeInTheDocument();
    expect(screen.getByText(`$${taxes.toFixed(2)}`)).toBeInTheDocument();
    expect(screen.getByText(`$${finalTotal.toFixed(2)}`)).toBeInTheDocument();
  });
});
