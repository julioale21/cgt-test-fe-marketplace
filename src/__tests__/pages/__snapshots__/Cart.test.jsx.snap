// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart should match snapshot with empty cart 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-l03i02-MuiPaper-root-MuiCard-root"
    style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiCardContent-root css-19iapi7-MuiCardContent-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-dhyiz8-MuiSvgIcon-root"
        data-testid="ShoppingCartIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2M1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2"
        />
      </svg>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-2qvmir-MuiTypography-root"
      >
        Your cart is empty
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-eicl6u-MuiButtonBase-root-MuiButton-root"
        id=":r0:"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1sh91j5-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="ArrowBackIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20z"
            />
          </svg>
        </span>
        Continue Shopping
      </button>
    </div>
  </div>
</div>
`;

exports[`Cart should match snapshot with items in cart 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-1m697zz-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 css-ngw6u5-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1j74uew-MuiPaper-root-MuiCard-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <div
          class="MuiCardHeader-root css-149tc0y-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content css-1jkp9bu-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-16xl4zq-MuiTypography-root"
            >
              <div
                class="MuiBox-root css-69i1ev"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root"
                >
                  Shopping Cart (
                  1
                   items)
                </h6>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-g4mrdg-MuiButtonBase-root-MuiButton-root"
                  id=":r1:"
                  tabindex="0"
                  type="button"
                >
                  Clear Cart
                </button>
              </div>
            </span>
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
        />
        <div
          class="MuiCardContent-root css-1lt5qva-MuiCardContent-root"
        >
          <div
            class="MuiStack-root css-1sazv7p-MuiStack-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1fdopfm-MuiPaper-root"
              style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
            >
              <div
                class="MuiBox-root css-j0ozid"
              >
                <img
                  alt="Test Product"
                  class="MuiBox-root css-npxvs5"
                  src="test.jpg"
                />
                <div
                  class="MuiBox-root css-i9gxme"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-subtitle1 css-1i3tsf3-MuiTypography-root"
                  >
                    Test Product
                  </h6>
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-1wle3ir-MuiTypography-root"
                  >
                    $
                    29.99
                  </p>
                </div>
                <div
                  class="MuiBox-root css-axw7ok"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-155wbr6-MuiButtonBase-root-MuiIconButton-root"
                    id=":r2:"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-loadingIndicator css-165vu30-MuiIconButton-loadingIndicator"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="RemoveIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 13H5v-2h14z"
                      />
                    </svg>
                  </button>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-owqrw6-MuiTypography-root"
                  >
                    2
                  </p>
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-155wbr6-MuiButtonBase-root-MuiIconButton-root"
                    id=":r3:"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-loadingIndicator css-165vu30-MuiIconButton-loadingIndicator"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="AddIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z"
                      />
                    </svg>
                  </button>
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeSmall css-195ve3w-MuiButtonBase-root-MuiIconButton-root"
                    id=":r4:"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-loadingIndicator css-165vu30-MuiIconButton-loadingIndicator"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="DeleteIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 css-18a7ljq-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-s7uqjd-MuiPaper-root-MuiCard-root"
        style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
      >
        <div
          class="MuiCardHeader-root css-149tc0y-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content css-1jkp9bu-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-16xl4zq-MuiTypography-root"
            >
              Order Summary
            </span>
          </div>
        </div>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
        />
        <div
          class="MuiCardContent-root css-1lt5qva-MuiCardContent-root"
        >
          <div
            class="MuiStack-root css-1sazv7p-MuiStack-root"
          >
            <div
              class="MuiBox-root css-gg4vpm"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-k16oju-MuiTypography-root"
              >
                Subtotal
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-4leci6-MuiTypography-root"
              >
                $59.98
              </p>
            </div>
            <div
              class="MuiBox-root css-gg4vpm"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-k16oju-MuiTypography-root"
              >
                Shipping
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-4leci6-MuiTypography-root"
              >
                Free
              </p>
            </div>
            <div
              class="MuiBox-root css-gg4vpm"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-k16oju-MuiTypography-root"
              >
                Estimated tax
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-4leci6-MuiTypography-root"
              >
                $6.00
              </p>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
            />
            <div
              class="MuiBox-root css-gg4vpm"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-vvtk0a-MuiTypography-root"
              >
                Total
              </h6>
              <h6
                class="MuiTypography-root MuiTypography-h6 css-vvtk0a-MuiTypography-root"
              >
                $65.98
              </h6>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-bks8m7-MuiButtonBase-root-MuiButton-root"
              id=":r5:"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1sh91j5-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="CreditCardIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 4H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V6c0-1.11-.89-2-2-2m0 14H4v-6h16zm0-10H4V6h16z"
                  />
                </svg>
              </span>
              Continue to Checkout
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-fullWidth css-acx9xf-MuiButtonBase-root-MuiButton-root"
              id=":r6:"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1sh91j5-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="ArrowBackIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20z"
                  />
                </svg>
              </span>
              Continue Shopping
            </button>
            <div
              class="MuiBox-root css-1yuhvjn"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-gutterBottom css-10f82c-MuiTypography-root"
              >
                Secure Checkout
              </h6>
              <div
                class="MuiStack-root css-hbbo2v-MuiStack-root"
              >
                <img
                  alt="Visa"
                  class="MuiBox-root css-tv7otk"
                  src="https://cdn.visa.com/v2/assets/images/logos/visa/blue/logo.png"
                />
                <img
                  alt="Mastercard"
                  class="MuiBox-root css-tv7otk"
                  src="https://cdn.worldvectorlogo.com/logos/mastercard-7.svg"
                />
                <img
                  alt="PayPal"
                  class="MuiBox-root css-tv7otk"
                  src="https://www.paypalobjects.com/digitalassets/c/website/logo/full-text/pp_fc_hl.svg"
                />
              </div>
            </div>
            <div
              class="MuiBox-root css-1yuhvjn"
            >
              <p
                class="MuiTypography-root MuiTypography-body2 css-yj4wuh-MuiTypography-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-120dh41-MuiSvgIcon-root"
                  data-testid="LocalShippingIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 8h-3V4H3c-1.1 0-2 .9-2 2v11h2c0 1.66 1.34 3 3 3s3-1.34 3-3h6c0 1.66 1.34 3 3 3s3-1.34 3-3h2v-5zM6 18.5c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5m13.5-9 1.96 2.5H17V9.5zm-1.5 9c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5"
                  />
                </svg>
                Free shipping on all orders
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
